src/Compilers/CSharp/Portable/Symbols/Metadata/PE/DynamicTypeDecoder.cs

```diff
@@ -338,6 +341,92 @@ private PointerTypeSymbol TransformPointerType(PointerTypeSymbol pointerType)
                 new PointerTypeSymbol(pointerType.PointedAtTypeWithAnnotations.WithTypeAndModifiers(transformedPointedAtType, pointerType.PointedAtTypeWithAnnotations.CustomModifiers));
         }
 
+#nullable enable
+        private FunctionPointerTypeSymbol? TransformFunctionPointerType(FunctionPointerTypeSymbol type)
+        {
+            var flag = ConsumeFlag();
+            Debug.Assert(!flag);
+
+            var sig = type.Signature;
+
+            var transformedReturn = handle(ref this, sig.RefKind, sig.RefCustomModifiers, sig.ReturnTypeWithAnnotations);
+            if (transformedReturn is null)
+            {
+                return null;
+            }
+
+            bool madeChanges = false;
+            TypeWithAnnotations transformedReturnWithAnnotations;
+
+            if (!transformedReturn.Equals(sig.ReturnType, TypeCompareKind.ConsiderEverything))
+            {
+                transformedReturnWithAnnotations = sig.ReturnTypeWithAnnotations.WithType(transformedReturn);
+                madeChanges = true;
+            }
+            else
+            {
+                transformedReturnWithAnnotations = sig.ReturnTypeWithAnnotations;
+            }
```

Comment:
Can `handle` transform `TypeWithAnnotations` instead of `TypeSymbol` to avoid duplicating this here and for parameters?